lib/backend.js

```diff
@@ -155,6 +166,13 @@ Backend.prototype.connect = function(connection, req) {
   // not used internal to ShareDB, but it is handy for server-side only user
   // code that may cache state on the agent and read it in middleware
   connection.agent = agent;
+
+  // Expose the DocPresence passed in through the constructor
+  // to the Doc class, which has access to the connection.
+  connection.DocPresence = this.presence.DocPresence;
+
+  connection._connectionPresence = new this.presence.ConnectionPresence(connection);
```

Comment:
As I stated above, I don't think this looks "normal". I'd possibly expect something more along the lines of:

```javascript
presence = this.presence.connectionPresence(connection);
```